lib/cli/lib/add.js

```diff
@@ -90,18 +90,34 @@ export const addStorybookAddonToFile = (addonName, addonsFile, isOfficialAddon)
   ];
 };
 
-const registerAddon = (addonName, isOfficialAddon) => {
-  const registerDone = commandLog(`Registering the ${addonName} Storybook addon`);
-  const addonsFilePath = path.resolve('.storybook/addons.js');
-  const addonsFile = fs
-    .readFileSync(addonsFilePath)
-    .toString()
-    .split('\n');
-  fs.writeFileSync(
-    addonsFilePath,
-    addStorybookAddonToFile(addonName, addonsFile, isOfficialAddon).join('\n')
-  );
-  registerDone();
+const postinstallAddon = async (addonName, isOfficialAddon) => {
+  let skipMsg = null;
+  if (!isOfficialAddon) {
+    skipMsg = 'unofficial addon';
+  } else {
+    const presetsCodemod = require.resolve(
+      `${getPackageName(addonName, isOfficialAddon)}/postinstall/presets.js`
+    );
+    if (fs.existsSync(presetsCodemod)) {
+      if (fs.existsSync('.storybook')) {
+        commandLog(`Running postinstall script for ${addonName}`)();
+        const presetsFile = path.join('.storybook', 'presets.js');
+        if (!fs.existsSync(presetsFile)) {
+          fs.writeFileSync(presetsFile, '', 'utf8');
+        }
+        spawnSync('npx', ['jscodeshift', '-t', presetsCodemod, presetsFile], {
```

Comment:
So the postinstall is actually a "post-add-modemod". It's limited to be exactly a codemod for presets.js only?

I THINK presets.js can be a .ts file too.

Reply:
Nope AFAIK it can't